concordium-sdk/src/main/java/com/concordium/sdk/Client.java

```diff
@@ -303,6 +304,28 @@ public PeerStatistics getPeerStatistics(final boolean includeBootstrappers) {
         return PeerStatistics.parse(value);
     }
 
+    /**
+     * Get the IDs of the bakers registered in the given block.
+     * Note that this list is in general different from the bakers that are
+     * returned as part of [get_birk_parameters](Client::get_birk_parameters).
+     * The latter are the bakers that are eligible for baking at the time of
+     * the block.
+     * @param blockHash {@link Hash} of the block bakers are to be retrieved.
+     * @return Parsed {@link ImmutableList} of {@link BakerId}
+     * @throws Exception When the returned JSON is null.
+     */
+    public ImmutableList<BakerId> getBakerList(Hash blockHash) throws Exception {
```

Comment:
I think we should use the `BlockNotFoundException` in these situations: https://github.com/Concordium/concordium-java-sdk/blob/main/concordium-sdk/src/main/java/com/concordium/sdk/exceptions/BlockNotFoundException.java

Reply:
done